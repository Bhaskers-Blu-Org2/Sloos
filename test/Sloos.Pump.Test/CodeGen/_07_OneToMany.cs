// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT license.

using ApprovalTests.Reporters;
using Sloos.Pump.EntityFramework;
using Sloos.Pump.Test.EntityFramework;
using Sloos.Pump.Test.Samples;
using Xunit;

namespace Sloos.Pump.Test
{
    [UseReporter(typeof(DiffReporter))]
    public class _07_OneToMany
    {
        [Fact]
        public void Reverse()
        {
            var entityNamespace = new EntityNamespace("Fact");

            var sampleGeneratorFactory = new SampleGeneratorFactory(
                CodeFirstGen.OneToMany.Csdl(),
                CodeFirstGen.OneToMany.Ssdl(),
                CodeFirstGen.OneToMany.Msdl());

            var factory = new ReverseFactory(
                sampleGeneratorFactory,
                entityNamespace,
                "Context");

            ApprovalsExtensions.Verify(factory);
        }
    }
}
