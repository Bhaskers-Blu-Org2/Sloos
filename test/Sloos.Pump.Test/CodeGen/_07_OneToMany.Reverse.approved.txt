namespace Fact.Models
{
    public partial class Context : global::System.Data.Entity.DbContext
    {
        static Context()
        {
            global::System.Data.Entity.Database.SetInitializer<Context>(null);
        }

        public Context(string nameOrConnectionString)
            : base(nameOrConnectionString)
        {
        }

        public global::System.Data.Entity.DbSet<global::Fact.Models.Post> Posts { get; set; }
        public global::System.Data.Entity.DbSet<global::Fact.Models.Tag> Tags { get; set; }

        protected override void OnModelCreating(global::System.Data.Entity.DbModelBuilder modelBuilder)
        {
            modelBuilder.Configurations.Add(new global::Fact.Models.Mapping.PostMap());
            modelBuilder.Configurations.Add(new global::Fact.Models.Mapping.TagMap());
        }
    }
}


namespace Fact.Models
{
    public partial class Post
    {
        public Post()
        {
            this.Tags = new global::System.Collections.Generic.List<Tag>();
        }
        public global::System.Int32 ID { get; set; }
        public virtual global::System.Collections.Generic.ICollection<Tag> Tags { get; set; }
    }
}


namespace Fact.Models
{
    public partial class Tag
    {
        public global::System.Int32 ID { get; set; }
        public global::System.Nullable<global::System.Int32> Post_ID { get; set; }
        public virtual Post Post { get; set; }
    }
}


namespace Fact.Models.Mapping
{
    public class PostMap : global::System.Data.Entity.ModelConfiguration.EntityTypeConfiguration<Post>
    {
        public PostMap()
        {
            // Primary Key
            this.HasKey(x => x.ID);

            // Properties

            // Table & Column Mappings
            this.ToTable("Posts", "dbo");
            this.Property(x => x.ID).HasColumnName("ID");

            // Relationships
        }
    }
}

namespace Fact.Models.Mapping
{
    public class TagMap : global::System.Data.Entity.ModelConfiguration.EntityTypeConfiguration<Tag>
    {
        public TagMap()
        {
            // Primary Key
            this.HasKey(x => x.ID);

            // Properties

            // Table & Column Mappings
            this.ToTable("Tags", "dbo");
            this.Property(x => x.ID).HasColumnName("ID");
            this.Property(x => x.Post_ID).HasColumnName("Post_ID");

            // Relationships
            this.HasOptional(x => x.Post)
                .WithMany(x => x.Tags)
                .HasForeignKey(x => x.Post_ID);
        }
    }
}

